mnist/mnist_gcp.ipynb

```diff
@@ -0,0 +1,895 @@
+    "kind: TFJob\n",
+    "metadata:\n",
+    "  name: {train_name}  \n",
+    "spec:\n",
+    "  tfReplicaSpecs:\n",
+    "    Ps:\n",
+    "      replicas: {num_ps}\n",
+    "      template:\n",
+    "        metadata:\n",
+    "          annotations:\n",
+    "            sidecar.istio.io/inject: \"false\"\n",
+    "        spec:\n",
+    "          serviceAccount: default-editor\n",
+    "          containers:\n",
+    "          - name: tensorflow\n",
+    "            command:\n",
+    "            - python\n",
+    "            - /opt/model.py\n",
+    "            - --tf-model-dir={model_dir}\n",
+    "            - --tf-export-dir={export_path}\n",
+    "            - --tf-train-steps={train_steps}\n",
+    "            - --tf-batch-size={batch_size}\n",
+    "            - --tf-learning-rate={learning_rate}\n",
+    "            image: {image}\n",
+    "            workingDir: /opt\n",
+    "          restartPolicy: OnFailure\n",
+    "    Chief:\n",
+    "      replicas: 1\n",
+    "      template:\n",
+    "        metadata:\n",
+    "          annotations:\n",
+    "            sidecar.istio.io/inject: \"false\"\n",
+    "        spec:\n",
+    "          serviceAccount: default-editor\n",
+    "          containers:\n",
+    "          - name: tensorflow\n",
+    "            command:\n",
+    "            - python\n",
+    "            - /opt/model.py\n",
+    "            - --tf-model-dir={model_dir}\n",
+    "            - --tf-export-dir={export_path}\n",
+    "            - --tf-train-steps={train_steps}\n",
+    "            - --tf-batch-size={batch_size}\n",
+    "            - --tf-learning-rate={learning_rate}\n",
+    "            image: {image}\n",
+    "            workingDir: /opt\n",
+    "          restartPolicy: OnFailure\n",
+    "    Worker:\n",
+    "      replicas: 1\n",
+    "      template:\n",
+    "        metadata:\n",
+    "          annotations:\n",
+    "            sidecar.istio.io/inject: \"false\"\n",
+    "        spec:\n",
+    "          serviceAccount: default-editor\n",
+    "          containers:\n",
+    "          - name: tensorflow\n",
+    "            command:\n",
+    "            - python\n",
+    "            - /opt/model.py\n",
+    "            - --tf-model-dir={model_dir}\n",
+    "            - --tf-export-dir={export_path}\n",
+    "            - --tf-train-steps={train_steps}\n",
+    "            - --tf-batch-size={batch_size}\n",
+    "            - --tf-learning-rate={learning_rate}\n",
+    "            image: {image}\n",
+    "            workingDir: /opt\n",
+    "          restartPolicy: OnFailure\n",
+    "\"\"\"           "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create the training job\n",
+    "\n",
+    "* We could write the spec to a YAML file and then do `kubectl apply -f {FILE}`\n",
+    "* Since we are running in jupyter we will use the Kubernetes python client"
```

Comment:
It might be helpful to also add a comment that re: 'we are deploying this job into your user profile namespace'

Reply:
Done.